### ISSUE TITLE
What version of the plugin has the rolling goal?

### ISSUE THREAD

**Issue by ntrvic** [fabric8io/fabric8#6179]
I just tried to run ```mvn fabric8:rolling``` with version 3.0.33 and I see this:
```[ERROR] Could not find goal 'rolling' in plugin io.fabric8:fabric8-maven-plugin:3.0.33 among available goals build, deploy, help, push, resource, run, watch```
In what version is that functionality available?

Thanks,
Victor

**Comment by jgangemi** [comment]
@rhuss might you be able to quickly shed any light on this? need it for the day job.

**Comment by jimmidyson** [comment]
Can you use Kubernetes deployments instead of the custom rolling functionality?

**Comment by ntrvic** [comment]
yeah we could, but would be nice to deploy rolling updates right from maven.

**Comment by jimmidyson** [comment]
One thing missing from the rolling goal is rollback on failure, plus some more refined options around scaling up/down. I'd recommend deployments personally.

But if you want to use maven then try `fabric8:deploy -Dfabric8.rolling=true` - haven't tried it myself but seems to do what you want looking at the code.

**Comment by rhuss** [comment]
You need to use a 2.x version of the plugin, though. 3.x is a complete rewrite which does not yet provide all the functionality of 2.x (and might not all as some of the 2.x funcs are more ops related than build related). 

The lates 2.x vesion is 2.2.144. Please note that it is also configured differently.

**Comment by jgangemi** [comment]
thanks @rhuss!!

**Comment by rhuss** [comment]
np :)

I agree with @jimmidyson that rolling updates should be a server side feature, not a client feature, though.

**Comment by jgangemi** [comment]
we are still figuring out how we are going to integrate kubes so this helps.

**Comment by jstrachan** [comment]
If you use the 3.x maven plugin it defaults to using Deployment on k8s or DeploymentConfig on OpenShift - so 'mvn fabric8:run' is the equivalent of the 2.x 'mvn fabric8:rolling' goal. 

Incidentally if you use Jenkins pipelines, the 'kubernetesApply' function behaves similarly too
https://github.com/jenkinsci/kubernetes-pipeline-plugin/blob/master/devops-steps/readme.md
